src/ncp/ncp_spinel.cpp

```diff
@@ -354,20 +338,42 @@ void NcpSpinel::HandleValueIs(spinel_prop_key_t aKey, const uint8_t *aBuffer, ui
     return;
 }
 
-void NcpSpinel::HandleResponseForCommand(spinel_tid_t aTid, otError aError)
+otbrError NcpSpinel::HandleResponseForPropSet(spinel_tid_t      aTid,
+                                              spinel_prop_key_t aKey,
+                                              const uint8_t    *aData,
+                                              uint16_t          aLength)
 {
-    switch (mCmdTable[aTid])
+    OTBR_UNUSED_VARIABLE(aData);
+    OTBR_UNUSED_VARIABLE(aLength);
+
+    otbrError error = OTBR_ERROR_NONE;
+
```

Comment:
I think we can check the key before the `switch` to avoid check the key in each case:

```C++
     VerifyOrExit(aKey == mWaitingKeyTable[aTid], error = OTBR_ERROR_INVALID_STATE);
```

Reply:
This is not always the case. For some SetProperty, the key in spinel response is the same property. For example, host sends Set `SPINEL_PROP_NET_IF_UP` and the response is VALUE_IS `SPINEL_PROP_NET_IF_UP`. However, for some properties, the response is VALUE_IS `SPINEL_PROP_LAST_STATUS` to indicate if the set was successful. So we have to handle them differently. In existing code, the responses are all the same prop key.